woocommerce/woocommerce-gutenberg-products-block
# Prepare All Products block to accept extension inner blocks

Fixes the remaining subtasks of #979.

This PR allows extensions to register blocks, so they can be added as _All Products_ children.

### Screenshots

![image](https://user-images.githubusercontent.com/3616980/66848719-e09b5880-ef75-11e9-8ad7-1b17dbaf07e0.png)

### How to test the changes in this Pull Request:

1. I created an example extension to test it. To install it:
  * `git clone git@github.com:Aljullu/woocommerce-blocks-extension.git`
  * `git checkout inner-blocks-extension`
  * `npm i && npm run start`
  * Then, activate the extension in wp-admin.
2. Add an _All Products_ block to a post and enter the edit mode (pencil icon in the block toolbar).
3. Verify there is a new child block you can add named _Product Price Level_.
![image](https://user-images.githubusercontent.com/3616980/66909564-4129a480-f00d-11e9-9b66-63146a27ad5a.png)
4. Add it and verify it renders correctly.
5. Preview the block and verify it renders correctly in the frontend too.
6. Deactivate the extension and verify the post still looks good both in the editor and the frontend.
7. Re-enable the extension and verify the block is there again.
